pkg/kubectl/cmd/set/set_entrypoint.go

```diff
@@ -0,0 +1,286 @@
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/google/shlex"
+	"github.com/spf13/cobra"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+// EntrypointOptions is the start of the data required to perform the operation.  As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type EntrypointOptions struct {
+	resource.FilenameOptions
+
+	Mapper       meta.RESTMapper
+	Infos        []*resource.Info
+	Encoder      runtime.Encoder
+	Decoder      runtime.Decoder
+	Selector     string
+	Out          io.Writer
+	Err          io.Writer
+	DryRun       bool
+	ShortOutput  bool
+	All          bool
+	Record       bool
+	Output       string
+	ChangeCause  string
+	Local        bool
+	Cmd          *cobra.Command
+
+	PrintSuccess           func(mapper meta.RESTMapper, shortOutput bool, out io.Writer, resource, name string, dryRun bool, operation string)
+	PrintObject            func(cmd *cobra.Command, isLocal bool, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error
+	UpdatePodSpecForObject func(obj runtime.Object, fn func(*v1.PodSpec) error) (bool, error)
+	Resources              []string
+	Container              string
+	Command                string
+	Args                   string
+	CommandSlice           []string
+	ArgSlice               []string
+}
+
+var (
+	entrypoint_long = templates.LongDesc(`
+		Update existing container entrypoint.`)
+
+	entrypoint_example = templates.Examples(`
+		# Update nginx-deployment container entrypoint command to '/bin/sh'
+		kubectl set entrypoint deployment nginx-deployment --container=nginx --command="/bin/sh ls -la" --args="test"`)
+)
+
+func NewCmdEntrypoint(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	options := &EntrypointOptions{
+		Out: out,
+		Err: err,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "entrypoint (-f FILENAME | TYPE NAME) CONTAINER_NAME --command='CMD_1 ... CMD_N' --args='ARG_1 ... ARG_N'",
+		Short:   i18n.T("Update entrypoint of a pod template"),
+		Long:    entrypoint_long,
+		Example: entrypoint_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "identifying the resource to get from a server."
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+	cmd.Flags().StringVar(&options.Command, "command", options.Command, "The command to be set for this container.  For example, \"/bin/sh 'ls -la'\".")
+	cmd.Flags().StringVar(&options.Args, "args", options.Args, "The command arguments for this container.  For example, 'test'.")
```

Comment:
You could use an example that shows that the "arguments" will also be spllitted.